spec/lib/reporting/protocols_report_spec.rb

```diff
@@ -113,6 +113,38 @@
     it 'generates the tabular csv data' do
       expect(report.as_tables).to eq expected_tables
     end
+
+    describe 'queries' do
+      let(:client) { report.cloudwatch_client }
+      let(:time_query) do
+        {
+          from: report.time_range.begin,
+          to: report.time_range.end,
+        }
+      end
+      before do
+        allow(client).to receive(:fetch).and_call_original
+      end
+
+      it 'calls the cloudwatch client with the expected queries' do
+        report.as_tables
+
+        %i[
+          aal3_issuers_query
+          saml_signature_query
+          facial_match_issuers_query
+          id_token_hint_query
+          loa_issuers_query
+          protocol_query
+          saml_signature_query
+        ].each do |query|
+          expect(client).to have_received(:fetch).with(
+            query: report.send(query),
```

Comment:
Nit: I think it'd be reasonable to assume (and assert) these methods are part of the class's public API?

```suggestion
            query: report.public_send(query),
```

Reply:
updated in https://github.com/18F/identity-idp/pull/11528/commits/67530e6af02e82fc0934b39b9e26c13cfc5ee1fc

Reply:
Yeah, my thought process was:

- "Are we calling internal methods in this spec in a way which pierces the abstraction?
- "Oh, no we're not, because those methods are indeed public"
- "So maybe we should add some guarantees that those are indeed public"
- "`public_send` will loudly fail if they're not public, and would be easy to swap in"

Reply:
thanks!